src/main/java/com/synopsys/integration/alert/provider/blackduck/web/PolicyNotificationFilterCustomFunctionAction.java

```diff
@@ -138,16 +138,14 @@ private boolean isFilterablePolicy(Collection<String> notificationTypes) {
 
     private Optional<BlackDuckProperties> createBlackDuckProperties(FieldModel fieldModel) throws IntegrationException {
         FieldAccessor fieldAccessor = fieldModelConverter.convertToFieldAccessor(fieldModel);
-        Optional<ConfigurationModel> configurationModel = configurationAccessor.getProviderConfigurationByName(fieldAccessor.getStringOrNull(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME));
-        Optional<FieldAccessor> providerFieldAccessor = configurationModel
-                                                            .map(ConfigurationModel::getCopyOfKeyToFieldMap)
-                                                            .map(FieldAccessor::new);
-
-        if (configurationModel.isPresent() && providerFieldAccessor.isPresent()) {
-            ConfigurationModel configModel = configurationModel.get();
-            return Optional.of(blackDuckPropertiesFactory.createProperties(configModel.getConfigurationId(), providerFieldAccessor.get()));
+        Long providerConfigId = fieldAccessor.getLong(ProviderDescriptor.KEY_PROVIDER_CONFIG_ID).orElse(null);
+        if (null == providerConfigId) {
+            return Optional.empty();
         }
-        return Optional.empty();
+        return configurationAccessor.getConfigurationById(providerConfigId)
```

Comment:
Just something to note, if we successfully remove AlertDatabaseConstraintException we'll be able to collapse this call into the previous Optional.